libtiff/tests/test_libtiff_ctypes.py

```diff
@@ -14,3 +16,562 @@ def test_issue69(tmp_path):
+        assert len(tmp[2]) == 256, \
+            f"Tag 'ColorMap' should be three arrays of 256 elements, found {len(tmp[2])} elements"
+    except TypeError:
+        print("Tag 'ColorMap' has the wrong shape of 3 arrays of 256 elements each")
+        return
+
+    print("Tag Read: SUCCESS")
+
+
+def test_write():
+    tiff = lt.TIFF.open('/tmp/libtiff_test_write.tiff', mode='w')
+    arr = np.zeros((5, 6), np.uint32)
+    for _i in range(arr.shape[0]):
+        for j in range(arr.shape[1]):
+            arr[_i, j] = _i + 10 * j
+    print(arr)
+    tiff.write_image(arr)
+    del tiff
+
+
+def test_read(filename=None):
+    filename = '/tmp/libtiff_test_write.tiff'
+
+    print('Trying to open', filename, '...', end=' ')
+    tiff = lt.TIFF.open(filename)
+    print('Trying to show info ...\n', '-' * 10)
+    print(tiff.info())
+    print('-' * 10, 'ok')
+    print('Trying show images ...')
+    t = time.time()
+    i = 0
+    for image in tiff.iter_images(verbose=True):
+        print(image.min(), image.max(), image.mean())
+        i += 1
+    print('\tok', (time.time() - t) * 1e3, 'ms', i, 'images')
+
+
+def test_write_float():
+    tiff = lt.TIFF.open('/tmp/libtiff_test_write.tiff', mode='w')
+    arr = np.zeros((5, 6), np.float64)
+    for _i in range(arr.shape[0]):
+        for j in range(arr.shape[1]):
+            arr[_i, j] = _i + 10 * j
+    print(arr)
+    tiff.write_image(arr)
+    del tiff
+
+    tiff = lt.TIFF.open('/tmp/libtiff_test_write.tiff', mode='r')
+    print(tiff.info())
+    arr2 = tiff.read_image()
+    print(arr2)
+
+
+def test_write_rgba():
+    tiff = lt.TIFF.open('/tmp/libtiff_test_write.tiff', mode='w')
+    arr = np.zeros((5, 6, 4), np.uint8)
+    for i in np.ndindex(*arr.shape):
+        arr[i] = 20 * i[0] + 10 * i[1] + i[2]
+    print(arr)
+    tiff.write_image(arr, write_rgb=True)
+    del tiff
+
+    tiff = lt.TIFF.open('/tmp/libtiff_test_write.tiff', mode='r')
+    print(tiff.info())
+    arr2 = tiff.read_image()
+    print(arr2)
+
+    np.testing.assert_array_equal(arr, arr2)
+
+
+def test_tree():
+    # Write a TIFF image with the following tree structure:
+    # Im0 --SubIFD--> Im0,1 ---> Im0,2 ---> Im0,3
+    #  |
+    #  V
+    # Im1
+    tiff = lt.TIFF.open('/tmp/libtiff_test_write.tiff', mode='w')
+    arr = np.zeros((5, 6), np.uint32)
+    for i in np.ndindex(*arr.shape):
+        arr[i] = i[0] + 20 * i[1]
+    print(arr)
+    n = 3
+    tiff.SetField("SubIFD", [0] * n)
+    tiff.write_image(arr)
+    for i in range(n):
+        arr[0, 0] = i
+        tiff.write_image(arr)
+
+    arr[0, 0] = 255
+    tiff.write_image(arr)
+    del tiff
+
+    tiff = lt.TIFF.open('/tmp/libtiff_test_write.tiff', mode='r')
+    print(tiff.info())
+    n = 0
+    for im in tiff.iter_images(verbose=True):
+        print(im)
+        n += 1
+
+    assert n == 2
+
+
+def test_copy():
+    tiff = lt.TIFF.open('/tmp/libtiff_test_compression.tiff', mode='w')
```

Comment:
You could use the `tmp_path` fixture since we're using pytest. Then every test execution would get its own directory that would be platform specific.

Reply:
Thanks for the tip. I've had to modify it a tiny bit to use the same folder in all the test cases, because some test cases write data, which is read by other test cases. But essentially this works nicely.

Reply:
How is that possible? I don't think pytest guarantees an ordering to test execution.

Reply:
Re the ordering of running tests w/ pytest: I think the order is undefined, I would assume it uses the add-to-globals-dict-ordering. However, there is https://pytest-ordering.readthedocs.io/en/develop/ .

That said, I think the most robust approach is the one used in `test_custom_tags`.

Reply:
I think I agree. That is, that any reading tests should also be doing the writing. Some of the writing tests look like they could be put into a separate function for creation or could be a pytest fixture that produces the filename of the written file. It could even be session or module scoped so it doesn't get recreated every test. Some of the tests like the tiled ones seem like they would benefit from `pytest.mark.parametrize` as they are only trying different ways of writing the data and making sure that the correct number of bytes is written.

Reply:
Let's keep it simple: it is ok if different tests write files with the same content. The gains of minimizing the number of files are almost none as there are not many files.

Reply:
Thanks for the recommendations. I've changed the test cases which relied on other test cases to call them first.
That means that some test cases are called more than once. In practice, that only seems to add a couple of seconds to the complete execution time, and allows to keep the code simple.